src/odemis/util/conversion.py

```diff
@@ -392,3 +394,78 @@ def ensure_tuple(v):
     else:
         return v
 
+def get_img_transformation_matrix(md):
+    """
+    Computes the 2D transformation matrix based on the given metadata.
+    md (dict str -> value): the metadata (of the DataArray) containing MD_PIXEL_SIZE 
+        and possibly also MD_ROTATION and MD_SHEAR.
+    return (numpy.matrix of 2,2 floats): the 2D transformation matrix
+    """
+
+    if model.MD_PIXEL_SIZE not in md:
+        raise ValueError("MD_PIXEL_SIZE must be set")
+    ps = md[model.MD_PIXEL_SIZE]
+    rotation = md.get(model.MD_ROTATION, 0.0)
+    shear = md.get(model.MD_SHEAR, 0.0)
+
+    # Y pixel coordinates goes down, but Y coordinates in world goes up
+    # The '-' before ps[1] is there to make this conversion
+    ps_mat = numpy.matrix([[ps[0], 0], [0, -ps[1]]])
+    cos, sin = numpy.cos(rotation), numpy.sin(rotation)
+    rot_mat = numpy.matrix([[cos, -sin], [sin, cos]])
+    shear_mat = numpy.matrix([[1, 0], [-shear, 1]])
+    return rot_mat * shear_mat * ps_mat
+
+def get_tile_md_pos(i, tile_size, tileda, origda):
+    """
+    Compute the position of the center of the tile, aka MD_POS.
+    i (int, int): the tile index (X, Y)
+    tile_size (int>0, int>0): the standard size of a tile in the (X, Y)
+    tileda (DataArray): the tile data, with MD_PIXEL_SIZE in its metadata.
+        It can be smaller than the tile_size in case
+    origda (DataArray or DataArrayShadow): the original/raw DataArray. If
+        no MD_POS is provided, the image is considered located at (0,0).
+    return (float, float): the center position
+    """
+    md = origda.metadata
+    tile_md = tileda.metadata
+    md_pos = numpy.asarray(md.get(model.MD_POS, (0.0, 0.0)))
+    if model.MD_PIXEL_SIZE not in md or model.MD_PIXEL_SIZE not in tile_md:
+        raise ValueError("MD_PIXEL_SIZE must be set")
+    orig_ps = numpy.asarray(md[model.MD_PIXEL_SIZE])
+    tile_ps = numpy.asarray(tile_md[model.MD_PIXEL_SIZE])
+
+    dims = md.get(model.MD_DIMS, "CTZYX"[-origda.ndim::])
+    img_shape = [origda.shape[dims.index('X')], origda.shape[dims.index('Y')]]
+    img_shape = numpy.array(img_shape, numpy.float)
+    # TODO check if the line below should be this way
+    # generate the shape of the zoomed image
+    img_shape *= orig_ps / tile_ps
+    # center of the image in pixels
+    img_center = img_shape / 2
+
+    tile_height = tileda.shape[dims.index('Y')]
+    tile_width = tileda.shape[dims.index('X')]
+    # center of the tile in pixels
+    tile_center_pixels = numpy.array([
+        i[0] * tile_size[0] + tile_width/2,
+        i[1] * tile_size[1] + tile_height/2]
+    )
+    # center of the tile relative to the center of the image
+    tile_rel_to_img_center_pixels = tile_center_pixels - img_center
+
+    # calculate the transformation matrix
+    tmat = get_img_transformation_matrix(tile_md)
```

Comment:
Probably, it should be equivalent, to multiply tile_center_pixels * (tile_ps/orig_ps), and to calculate the transformation matrix on the md. It would make the code a little more obvious on what it's doing.